contracts/token/ERC721/ERC721Metadata.sol

```diff
@@ -56,7 +59,16 @@ contract ERC721Metadata is Context, ERC165, ERC721, IERC721Metadata {
      * Throws if the token ID does not exist. May return an empty string.
      * @param tokenId uint256 ID of the token to query
      */
-    function tokenURI(uint256 tokenId) external view returns (string memory) {
+    function tokenURI(uint256 tokenId) public view returns (string memory) {
+        return string(abi.encodePacked(_baseTokenURI(), _tokenURI(tokenId)));
```

Comment:
For the sake of readability, we may want to add a `concat` function in the future

Reply:
We've discussed this in the past, in fact in #1746, and decided not to provide a `concat` function. The argument was that `string` is variadic and our function would have a fixed number of arguments, and potentially lead to worse performance than is possible by using `string`.

Reply:
Given that readability is rather important, I wouldn't be against having variants with different arities (say up to 5), though it is somewhat hacky.